### PR TITLE
Re-enable Epoch Integration Tests

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR re-enables all epoch integration tests, reduces time taken by epoch integration tests by about 30%, and fixes a bug in the snapshot validator function.

- Fix off-by-one error in snapshot validator function
- Update documentation of validator, and require in Access Node as well
- Remove expiry-escape mitigation in AN ingestion, which is now covered by the snapshot validator
- Replace separate staking step transactions (create account, fund, set up staking collection, register node) with one custom transaction
- This enables reducing the epoch length by 50%, speeding up the test significantly

#### Local 
Before improvements, was 650-720s.
After improvements, is 400-430s
#### CI
Before improvements, was 30-40m
After improvements, is ~23m

### CODE REVIEW

File: integration/tests/epochs/suite.go [hunk 1]
```diff
@@ -168,61 +168,59 @@ type StakedNodeOperationInfo struct {
 // 4. Add additional funds to staking account for storage
 // 5. Create Staking collection for node
 // 6. Register node using staking collection object
+// 7. Add the node to the approved list
+//
 // NOTE: assumes staking occurs in first epoch (counter 0)
+// NOTE 2: This function performs steps 1-7 in one custom transaction, to reduce
```
Comment (jordanschalm) [inline integration/tests/epochs/suite.go:174]:
```suggestion
// NOTE 2: This function performs steps 1-6 in one custom transaction, to reduce
```

**Review by durkmurder — APPROVED** [review]
Looks good

**Review by AlexHentschel — APPROVED** [review]
very nice.

**Comment by jordanschalm** [comment]
bors merge

**Pull request merged**